tests/Doctrine/Tests/DBAL/Functional/Driver/SQLAnywhere/StatementTest.php

```diff
@@ -50,4 +51,54 @@ public function testPersistentStatement()
         self::assertTrue($prepStmt->execute(),' Statement persistent failed');
     }
 
+    /**
+     * Check Issue #2991
+     * @throws \Doctrine\DBAL\ConnectionException
+     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
+     */
+    public function testBindParams()
+    {
+        $params = $this->_conn->getParams();
+
+        $conn = DriverManager::getConnection($params);
+
+        $conn->connect();
+        $conn->beginTransaction();
+        try {
```

Comment:
I'd avoid having flow control structures in tests as much as possible. All side effects should be handled in `tearDown()`.

Reply:
removed and createTable/dropTable moved to setUp/tearDown